File: src/clase-05/practica/typescript/activity4_facade_starter/facade.ts
import { Database, Auth, Logger } from "./subsystems";

export class SystemFacade {
  private db = new Database();
  private auth = new Auth();
  private logger = new Logger();

  startApp(user: string): string {
    const steps = [
      this.logger.start(),
      this.db.connect(),
      this.auth.init(),
      this.auth.login(user),
      this.db.query("SELECT 1"),
      this.logger.log("Sistema inicializado"),
    ];
    return steps.join(" | ");
  }

  shutdown(): string {
    const steps = [
      this.logger.stop(),
      this.auth.shutdown(),
      this.db.close(),
    ];
    return steps.join(" | ");
  }
}

File: src/clase-05/practica/python/activity4_facade_starter/facade.py
from subsystems import Database, Auth, Logger

class SystemFacade:
    def __init__(self):
        self.db = Database()
        self.auth = Auth()
        self.logger = Logger()

    def start_app(self, user: str) -> str:
        steps = [
            self.logger.start(),
            self.db.connect(),
            self.auth.init(),
            self.auth.login(user),
            self.db.query("SELECT 1"),
            self.logger.log("Sistema inicializado"),
        ]
        return " | ".join(steps)

    def shutdown(self) -> str:
        steps = [
            self.logger.stop(),
            self.auth.shutdown(),
            self.db.close(),
        ]
        return " | ".join(steps)
